client/src/components/profile/components/internet.tsx

```diff
@@ -107,46 +90,38 @@ class InternetSettings extends Component<InternetProps, InternetState> {
       state: 'error',
       message: t('validation.invalid-url')
     };
-  }
+  };
 
-  createHandleChange =
+  const createHandleChange =
     (key: keyof Socials) => (e: React.FormEvent<HTMLInputElement>) => {
       const value = (e.target as HTMLInputElement).value.slice(0);
-      return this.setState(state => ({
-        formValues: {
-          ...state.formValues,
-          [key]: value
-        }
-      }));
+      setFormValues({
+        ...formValues,
+        [key]: value
+      });
     };
 
-  isFormPristine = () => {
-    const { formValues, originalValues } = this.state;
+  const isFormPristine = () => {
     return (Object.keys(originalValues) as Array<keyof Socials>).every(
       key => originalValues[key] === formValues[key]
     );
   };
 
-  isFormValid = (): boolean =>
-    Object.values(this.state.formValues).every(
+  const isFormValid = (): boolean =>
+    Object.values(formValues).every(
       (value: string) => value === '' || isURL(value)
     );
 
-  handleSubmit = (e: React.FormEvent) => {
+  const handleSubmit = (e: React.FormEvent) => {
     e.preventDefault();
-    if (!this.isFormPristine() && this.isFormValid()) {
-      // // Only submit the form if is has changed, and if it is valid
-      const { formValues } = this.state;
-
-      const { updateSocials } = this.props;
-      this.toggleEditing();
-      return updateSocials({ ...formValues });
+    if (!isFormPristine() && isFormValid()) {
+      // Only submit the form if is has changed, and if it is valid
+      updateSocials({ ...formValues });
     }
-    this.toggleEditing();
-    return null;
+    setIsEditing(false);
   };
 
-  renderCheck = (
+  const renderCheck = (
```

Comment:
While were refactoring, could you convert this into a component (defined outside `InternetSettings`)?

Functions that return JSX can and do work, but they are a trap. Reason being, they look so much like components that people call hooks inside them. This does work well.

We've been bitten by this before and it was a pain to debug!

Reply:
Sure!